### ISSUE TITLE
test-finalization-group-error failed on v8 8.4

### ISSUE THREAD

**Issue by gengjiawen** [nodejs/node-v8#156]
This one still failed:
cc @mmarchini  
```console
=== release test-finalization-group-error ===
Path: parallel/test-finalization-group-error
--- stdout ---
Mismatched noop function calls. Expected exactly 1, actual 2.
    at Proxy.mustCall (/home/runner/work/node-v8/node-v8/test/common/index.js:328:10)
    at Object.<anonymous> (/home/runner/work/node-v8/node-v8/test/parallel/test-finalization-group-error.js:27:40)
    at Module._compile (internal/modules/cjs/loader.js:1178:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1198:10)
    at Module.load (internal/modules/cjs/loader.js:1027:32)
    at Function.Module._load (internal/modules/cjs/loader.js:923:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:71:12)
    at internal/main/run_main_module.js:17:47
Command: out/Release/node --expose-gc --harmony-weak-refs /home/runner/work/node-v8/node-v8/test/parallel/test-finalization-group-error.js
```

_Originally posted by @gengjiawen in https://github.com/nodejs/node-v8/issues/155#issuecomment-613173121_

**Comment by gengjiawen** [comment]
I debugged this, throw is triggered, but `uncaughtException` not noticed. 

https://github.com/nodejs/node-v8/blob/e1230d518d2dc28feafacab58d644583f7c78406/test/parallel/test-finalization-group-error.js#L8-L10

cc @devsnek @syg

**Comment by syg** [comment]
Pretty sure this is because you need to change the flag to `--harmony-weak-refs-with-cleanup-some`. At the last TC39, Apple objected to `cleanupSome`, so the method was made optional. V8 split out `cleanupSome` to a different flag so the rest of the API may ship. That test uses `cleanupSome`, which is no longer exposed by default.

**Comment by gengjiawen** [comment]
I am aware of that, when I debug, I add `--harmony-weak-refs-with-cleanup-some`, but the test still failed.

**Comment by devsnek** [comment]
@syg could recent changes (maybe the torque rewrite?) have broken the uncaught exception reporting? there's been a similar issue with microtasks for a while.

**Comment by syg** [comment]
I debugged this locally. I noticed three issues with node's WeakRef integration. I'm not exactly sure how this test worked before.

1. The top-level line `g.register({}, 42)` doesn't actually cause the unreachable `{}` to be GC-able when calling `gc()`, so the async task wasn't being scheduled and wasn't running at all. I didn't dig into why exactly, but other V8 tests tend to do the registration in an IIFE, like `(function () { g.register({}, 42); })();`, which is a more foolproof for causing the object to be collected.

1. There's a single object registered, and `FinalizationRegistry`'s API has changed to a per-item API. `cleanupSome` will drain all finalized objects, so if `cleanupSome` is called, no async task will be scheduled even if the scheduling was working.

1.  As per the the deprecation message for `Isolate::SetHostCleanupFinalizationGroupCallback`, the scheduling of `FinalizationRegistry` cleanup tasks is automatic if no host cleanup callback is set. I see that this version of node is still using the hook. Things should "just work" if no hook is set.

Testing weak refs and finalizers is hard, good luck!
